### ISSUE TITLE
Add Python 3.10 (release)

### ISSUE THREAD

**Issue by nolar** [actions/virtual-environments#4226]
### Tool name

Python

### Tool license

PSF

### Add or update?

- [ ] Add
- [X] Update

### Desired version

3.10 (released yesterday)

### Virtual environments affected

- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] Windows Server 2016
- [ ] Windows Server 2019
- [ ] Windows Server 2022

**Comment by mikhailkoliada** [comment]
Hello @nolar!
Thank you for the report, we'll take a closer look at this issue.

**Comment by hugovk** [comment]
For reference, it took 22 days for Python 3.9 (https://github.com/actions/virtual-environments/issues/1740) and 24 days for Python 3.8 (https://web.archive.org/web/20201012044438/https://github.com/microsoft/azure-pipelines-image-generation/issues/1317) :)

**Comment by miketimofeev** [comment]
@nolar @hugovk Python 3.10 is already available through https://github.com/actions/setup-python and will be pre-cached on the next week's images.

**Comment by hugovk** [comment]
Thank you! Confirmed now available for GitHub Actions: https://github.com/actions/setup-python/issues/249#issuecomment-934193641

---

Does Azure Pipelines need to wait for the pre-cached images next week?

```
Starting: UsePythonVersion
==============================================================================
Task         : Use Python version
Description  : Use the specified version of Python from the tool cache, optionally adding it to the PATH
Version      : 0.188.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/tool/use-python-version
==============================================================================
##[error]Version spec 3.10 for architecture x64 did not match any version in Agent.ToolsDirectory.
Versions in /opt/hostedtoolcache:

2.7.18 (x64)
3.5.10 (x64)
3.6.15 (x64)
3.7.12 (x64)
3.8.12 (x64)
3.9.7 (x64)
If this is a Microsoft-hosted agent, check that this image supports side-by-side versions of Python at https://aka.ms/hosted-agent-software.
If this is a self-hosted agent, see how to configure side-by-side Python versions at https://go.microsoft.com/fwlink/?linkid=871498.
Finishing: UsePythonVersion
```

https://dev.azure.com/hugovk/hugovk/_build/results?buildId=2488&view=logs&j=1c03d131-585d-5673-8b12-4599c17588ee

**Comment by nolar** [comment]
Indeed, it works! https://github.com/nolar/kopf/runs/3801486572?check_suite_focus=true

**Comment by hugovk** [comment]
Please could we re-open this to track Azure Pipelines? Or shall I open a new issue (here or elsewhere)?

**Comment by nolar** [comment]
Sorry. Reopened.

**Comment by miketimofeev** [comment]
@hugovk correct, AzDO task can't download the version on demand so we have to wait until the new images are available

**Comment by hugovk** [comment]
Short version: put quotes around the versions, so it interprets "3.10" as a string and not "3.1" as a float:

```yaml
    python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```

Longer version: https://dev.to/hugovk/the-python-3-1-problem-85g

**Comment by eccles** [comment]
ah thanx - alt=ready saw this when inspacting the commit here...

**Comment by charris** [comment]
@miketimofeev No problem with no 32 bit 3.10 on windows, we would prefer to drop it ourselves. As to the missing `Python.h`, it is simply a question of if it is installed and, if so, where.

**Comment by charris** [comment]
@miketimofeev  The problem looks to be a change in Python 3.10 distutils.

**Comment by miketimofeev** [comment]
@charris we have a feature request to support universal2 builds and a PR as well but didn't have a chance to validate it yet https://github.com/actions/python-versions/pull/114

**Comment by hugovk** [comment]
Thank you!

I count that as 17 days since Python 3.10, an improvement over last year's 22 days since 3.9 and the year before's 24 days since 3.8! 🎈👏
